pallets/did/src/lib.rs

```diff
@@ -857,6 +865,9 @@ pub mod pallet {
 				},
 			)?;
 			DidEndpointsCount::<T>::insert(&did_subject, currently_stored_endpoints_count.saturating_add(1));
+			did.update_deposit(&did_subject)?;
```

Comment:
In general, the `update_deposit` function must be called first, and then the record has to be saved. Otherwise, the deposit attribute will not be updated, and the state is not correct.

Reply:
In general, the `update_deposit` function must be called first. Otherwise, the deposit is not updated in the `did_details`.

Reply:
Same as above. The new service endpoint has to be in the storage, otherwise the update_deposit function would calculate the old deposit value.